crates/ruff_cli/src/lib.rs

```diff
@@ -329,10 +329,22 @@ pub fn check(args: CheckArgs, log_level: LogLevel) -> Result<ExitStatus> {
             }
         }
     } else {
-        let is_stdin = cli.files == vec![PathBuf::from("-")];
+        // If the user provided the dash file (`-`), or specified a `--stdin-filename`,
+        // read input from standard input.
+        let is_stdin =
+            cli.stdin_filename.is_some() || cli.files.iter().any(|file| file == Path::new("-"));
```

Comment:
How do other tools handle the case where you specify multiple file names including the `-`?

I'm asking because my expectation would be that ruff treats `-` as a file if you run `ruff a - b`. 

Overall, it's problematic that `-` is ambiguous and are leaning towards always requiring `--stdin-filename`.

Reply:
It looks like ESLint has a `--stdin` flag, we should... probably do that?